clang/include/clang/Basic/AttrDocs.td

```diff
@@ -1473,6 +1473,66 @@ Further reading for other annotations:
   }];
 }
 
+def CHERIMMIODeviceDocs : Documentation {
+  let Category = DocCatVariable;
+  let Content = [{
+Indicates that the extern volatile global it refers to must be treated as
+a cross-compartment object. In particular, this attribute indicates that the
+object is related to an MMIO-bound device.
+
+**Usage**: ``__attribute((cheriot_mmio("<device>",
+"<permissions_encoding>")))``, where ``"<device>"`` is the (quoted) identifier
+of the MMIO-bound device. An example of this is ``"uart"``. The
+``"<permissions_encoding>"`` indicates the permissions of the capability
+resulting from the import. The permissions encoding is a string of variable
+length. The symbols allowed in the string are ``R``, ``W``, ``c`` and ``m``,
+whose meaning is explained in the table below.
+
+
+.. csv-table:: Supported Syntaxes
+   :header: "Symbol", "Description", "Meaning", "Depends on"
+
+   "``R``","Load (Read)", "May be used to read.",""
+   "``W``","Store (Write)", "May be used to write.",""
+   "``c``","Load / Store Capability", "May be used to load or store capabilities as well as non-capability data.","``R`` or ``W``"
+   "``m``","Load Mutable", "May be used to load capabilities with write permission.","``R``"
+
+Examples of valid encodings are: ``"RWcm"`` (all permissions), ``"R"`` (read only), etc.
+
+**Warning**: The ``<permissions_encoding>`` parameter is optional, and if no encoding is given ``"RWcm"`` is assumed.
+  }];
+}
+
+
+def CHERISharedObjectDocs : Documentation {
+  let Category = DocCatVariable;
+  let Content = [{
+Indicates that the extern volatile global it refers to must be treated as
```

Comment:
If it the global must be volatile, then I would make it an explicitly stated requirement, i.e. "The indicated global must also have the `volatile` qualifier."

Reply:
[f4dad8b](https://github.com/CHERIoT-Platform/llvm-project/pull/147/commits/f4dad8b48da35cbe0f0a8468f8b2969b4541da9a) specifies that `extern` and `volatile` are mandatory (although currently not enforced) qualifiers.